1:1 error:Function foo2 already exists